quicktest_test.go

```diff
@@ -527,6 +528,26 @@ want:
 `)
 }
 
+// TestCRunGetFuncSig checks the internal function getRunFuncSignature which is used in Run.
+func TestCRunGetFuncSig(t *testing.T) {
+	for _, tb := range []testing.TB{
+		t,                 // *testing.T
+		(*testing.B)(nil), // *testing.B
+		qt.New(t),         // *quicktest.C
+		(*customT)(nil),   // a custom form of testing.TB with a non-standard Run method
+	} {
+		tbt := reflect.TypeOf(tb)
+		farg, err := qt.GetRunFuncSignature(tbt)
```

Comment:
This test isn't great IMHO. Firstly, it tests GetRunFuncSignature which is an internal implementation detail (this module in general tries to test the public API as much as possible to allow later internal refactoring without changing tests). Secondly, GetRunFuncSignature won't even be invoked in most of the above cases, so I'm not sure why this is explicitly testing them.

Reply:
You're right. I had written that test in the context of #160, but it isn't relevant anymore here.

Also, the extended tests and benchmarks you wrote in #166 would be welcome on a separate MR (written without generics) and I would be glad to rebase.

Reply:
I have now removed the commit that added that test.